sdk/identity/azure-identity/azure/identity/_credentials/managed_identity.py

```diff
@@ -17,6 +17,27 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def validate_identity_config(client_id: Optional[str], identity_config: Optional[Mapping[str, str]]) -> None:
+    if identity_config:
+        if client_id:
+            if any(key in identity_config for key in ("object_id", "resource_id")):
+                raise ValueError(
+                    "identity_config must not contain 'object_id' or 'resource_id' when 'client_id' is present"
+                )
+            if "client_id" in identity_config and identity_config["client_id"] != client_id:
```

Comment:
I don't think we need to validate if they are same or not. As long as client_id is specified in two places, we raise.